lib/modules/manager/gradle/utils.ts

```diff
@@ -13,7 +13,7 @@ const artifactRegex = regEx(
   '^[a-zA-Z][-_a-zA-Z0-9]*(?:\\.[a-zA-Z0-9][-_a-zA-Z0-9]*?)*$'
 );
 
-const versionLikeRegex = regEx('^(?<version>[-.\\[\\](),a-zA-Z0-9+]+)');
+const versionLikeRegex = regEx('^(?<version>[-_.\\[\\](),a-zA-Z0-9+]+)');
```

Comment:
```suggestion
const versionLikeRegex = regEx('^(?<version>[-.\\[\\](),a-zA-Z0-9+]+(_\\d+)*)*');
```

Reply:
Thanks for the suggestion. I'd argue against it:
- An underscore may occur at any position of the version, not necessarily **after** `-.\` and other characters
- Adding an additional matching group for _\d+ makes no sense here
- The trailing `*` would lead to multiple consecutive version matches, e.g. with a string like `1.3.0_456_1.3.0`  -> that's not desired behavior